api/oidc_test.go

```diff
@@ -0,0 +1,143 @@
+package api_test
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/buildkite/agent/v3/api"
+	"github.com/buildkite/agent/v3/logger"
+	"github.com/google/go-cmp/cmp"
+)
+
+func newOidcTokenServer(
+	t *testing.T,
+	accessToken, oidcToken, path string,
+	expectedBody []byte,
+) *httptest.Server {
+	t.Helper()
```

Comment:
Breaking function declarations across multiple lines feels weird to me. It's the sort of thing that leads to indentation-level confusion.
And many other Go folks think there's no problem with long lines anyway:
https://github.com/golang/go/wiki/CodeReviewComments#line-length

Reply:
I think this line would be “uncomfortably long” if the function declaration were not wrapped. It's a lot easier to read a function declaration that has been intentionally wrapped for each argument than one that has been arbitrarily wrapped by the code editor. So I prefer to do it this way.

Reply:
Okay - I won't hold you up with a debate around wrapping. It's not unreadable. I'll go away and think about it.